libraries/mbed/targets/hal/TARGET_NORDIC/TARGET_MCU_NRF51822/sleep.c

```diff
@@ -28,5 +28,5 @@ void sleep(void)
 void deepsleep(void)
 {
     sleep();
-    //   NRF_POWER->SYSTEMOFF=1;
+    NRF_POWER->SYSTEMOFF = 1;
```

Comment:
Setting this register to 1 powers off the SoC. You're right and I should have commented the call to ```sleep()```